### PR TITLE
Border erosion for RDNIM benchmarks

### PR DESCRIPTION
**Pull request by hardik01shah**
The RDNIM dataset contains strong homography transformations that lead to significant border artifacts in the warped images. This makes the point-line extractor susceptible to detecting keypoints and lines on the synthetic borders in the warped image. To prevent this, we calculate a mask in the dataloader that denotes all the valid pixels in the image that should be considered as keypoint candidates. We multiply the score_map generated by the extractor with the mask and then continue with NMS, keypoint detection and line detection.

### CODE REVIEW

File: gluefactory/datasets/rdnim.py [hunk 1]
```diff
@@ -77,7 +78,20 @@ def _init(self, conf):
 
     def _read_image(self, idx: int, type: str) -> dict:
         img = load_image(self._files[idx][type], self.conf.grayscale)
-        return self.preprocessor(img)
+
+        # get mask for black pixels in the image to identify synthetic borders
+        np_img = read_image(self._files[idx][type], self.conf.grayscale)
+        mask = ~np.all(np_img == 0, axis=2)
```
Comment (FelixMoeller3) [inline gluefactory/datasets/rdnim.py:84]: Can we rename the mask variable? It's a bit unexpressive

File: gluefactory/models/lines/pold2_extractor.py [hunk 2]
```diff
@@ -322,6 +322,8 @@ def filter_with_distance_field(
         points_coordinates = self.get_coordinates(
             points, indices, self.coeffs_list[sample_idx], self.coeffs_second_list[sample_idx]
         )
+        points_coordinates[:, 0] = torch.clamp(points_coordinates[:, 0], 0, distance_map.shape[1] - 1)
```
Comment (FelixMoeller3) [inline gluefactory/models/lines/pold2_extractor.py:325]: Why do we need the clamping here?

**Review by FelixMoeller3 — APPROVED** [review]

**Comment by FelixMoeller3** [comment]
@hardik01shah could you address my comments so we can merge this soon?

File: gluefactory/models/lines/pold2_extractor.py (hunk 2 above)
Comment (hardik01shah) [reply gluefactory/models/lines/pold2_extractor.py:325]: point_coordintates in rdnim sometimes fell exactly on the border, which led to indexing errors when indexing the df.

File: gluefactory/datasets/rdnim.py (hunk 1 above)
Comment (hardik01shah) [reply gluefactory/datasets/rdnim.py]: Done, changed to `padding_mask`
